main.go

```diff
@@ -89,13 +90,20 @@ func main() {
 		"The controller will load its initial configuration from this file. "+
 			"Omit this flag to use the default configuration values. ")
 
+	var featureGates string
+	flag.StringVar(&featureGates, "feature-gates", "", "A set of key=value pairs that describe feature gates for alpha/experimental features.")
+
 	opts := zap.Options{
 		TimeEncoder: zapcore.RFC3339NanoTimeEncoder,
 		ZapOpts:     []zaplog.Option{zaplog.AddCaller()},
 	}
 	opts.BindFlags(flag.CommandLine)
 	flag.Parse()
 
+	if err := utilfeature.DefaultMutableFeatureGate.Set(featureGates); err != nil {
+		os.Exit(1)
```

Comment:
Can we output an error log?

Reply:
Make sense, and I'll add another log error below the same time.: